### PR TITLE
[BD-6] OEP-18 compliance and python 3.8 support

### PR DESCRIPTION
**Pull request by ericfab179**
Update setup.py install_requirements definition.
Create requirements folder.
create pip_tools, base, test, tox and travis requirements files.
Add Makefile. Add makefile upgrade command.
Include requirements files generated using upgrade command.
Update openedx.yaml OEPs list.
Add tox and travis files.
Add MANIFEST.in.
Fix tests (str to bytes in hmac).

**Reviewers**
- [ ] @awais786 
- [ ] @morenol

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ericfab179! I've created [OSPR-4782](https://openedx.atlassian.net/browse/OSPR-4782) to keep track of it in JIRA, where we prioritize reviews.  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

Please let us know once your PR is ready for our review and all tests are green.

File: schoolyourself/schoolyourself_review.py [hunk 1]
```diff
@@ -125,18 +125,18 @@ def handle_grade_json(self, data):
 
       # Verify the signature.
       if isinstance(self.shared_key, text_type):
-          self.shared_key = self.shared_key.encode('utf-8')
-      verifier = hmac.new(str(self.shared_key), user_id)
+        sk = self.shared_key.encode('utf-8')
+      verifier = hmac.new(sk, user_id.encode('utf-8'), digestmod='MD5')
```
Comment (morenol) [inline schoolyourself/schoolyourself_review.py:129]: I think that `sk` will be undefined at this point if it does not enter in the previous if

File: MANIFEST.in [hunk 2]
```diff
@@ -0,0 +1 @@
+include requirements/base.in
```
Comment (morenol) [inline MANIFEST.in:1]: I think that we need to add the a recursive-include of schoolyourself folder.

File: schoolyourself/schoolyourself_review.py (hunk 1 above)
Comment (ericfab179) [reply schoolyourself/schoolyourself_review.py:130]: @morenol I fixed it assigning it before the if statement. Is it okay ?

**Pull request merged**